hera_cal/reflections.py

```diff
@@ -442,85 +832,194 @@ def fit_reflection(data, dly_range, freqs, full_freqs=None, edgecut=0,
     # imag = np.array([interp1d(x, y[i].imag, kind='linear')(bin_shifts[i]) for i in range(Ntimes)])
     # ref_phs = np.angle(real + 1j*imag)
 
-    # construct reflection
-    if full_freqs is None:
-        full_freqs = freqs
-    eps = construct_reflection(full_freqs, ref_amps, ref_dlys, ref_phs)
+    return ref_phs
 
-    return (eps, ref_amps, ref_dlys, ref_phs, ref_dly_inds, ref_significance,
-            np.fft.fftshift(dfft, axes=1), np.fft.fftshift(dlys))
 
+def _svd_waterfall(data):
+    """
+    Take a singular-value decomposition of a 2D
+    visibility in either frequency or delay space,
+    with shape (Ntimes, Nfreqs) or (Ntimes, Ndelays).
+
+    Parameters
+    ----------
+    data : 2d complex ndarray
+
+    Returns
+    -------
+    PCs : 3d ndarray
+        Holds principal components of the data
+        with shape=(Npcs, Ntimes, Nfreqs)
+
+    svals : 1d ndarray
+        Holds the singluar (or eigen) values of the
+        principal components.
+    """
+    # get singular values and eigenbases
+    u, svals, v = np.linalg.svd(data)
+
+    return u, svals, v
+
+
+def _form_PCs(u, v):
+    Nu = u.shape[1]
+    Nv = v.shape[0]
+    Npc = min([Nu, Nv])
 
-def reflections_delay_filter(data, flags, dnu, dly_cut=200, edgecut=0, taper='none', alpha=0.1, tol=1e-5, maxiter=500,
-                             gain=0.1, skip_wgt=0.2, timeavg=False, broadcast_flags=False, time_thresh=0.1):
+    # calculate outer products to get principal components
+    PCs = np.einsum("ij,jk->jik", u, v)
+
+    return PCs
+
+
+def sum_principal_components(svals, PCs, Nkeep=None):
+    """
+    Dot singular values into principal components,
+    keeping a specified number of PCs. svals should
+    be rank ordered from highest to lowest.
+
+    Parameters
+    ----------
+    svals : 1D ndarray
+        singular values, with length Nvals
+
+    PCs : 3D ndarray
```

Comment:
This doesn't make sense. You've artificially produced a much larger object in memory than you need. Do the outer products on the fly so you never need to store anything larger than the original data.